File: Java/src/main/java/QuestionBank/Solution1301_1400/Solution1379.java
package QuestionBank.Solution1301_1400;
/**
 * @author windows
 */
public class Solution1379 {

    public final TreeNode getTargetCopy(final TreeNode original, final TreeNode cloned, final TreeNode target) {
        if (original == null || original == null) return null;
        if (original == target) return cloned;
        TreeNode left = getTargetCopy(original.left, cloned.left, target);
        if (left != null) return left;
        return getTargetCopy(original.right, cloned.right, target);
    }
    
    
    public static void main(String[] args) {
        Solution1379 solution1379 = new Solution1379();
    }


    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;
        TreeNode(int x) { val = x; }
    }
}

File: C++/src/Solution1301_1400/Solution1379.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution1379.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

//Definition for a binary tree node.
struct TreeNode {
  int val;
  TreeNode *left;
  TreeNode *right;
  TreeNode() : val(0), left(nullptr), right(nullptr) {}
  TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
  TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

TreeNode* getTargetCopy(TreeNode* original, TreeNode* cloned, TreeNode* target) {
  if(original == nullptr || cloned == nullptr) return nullptr;
  if(original == target) return cloned;
  TreeNode* left = getTargetCopy(original->left, cloned->left, target);
  if(left != nullptr) return left;
  return getTargetCopy(original->right, cloned->right, target);
}